lib/methwhy.g

```diff
@@ -119,35 +196,48 @@ end);
 ##  <#/GAPDoc>
 ##
 BIND_GLOBAL("ApplicableMethodTypes",function(arg)
-local oper,narg,args,skip,verbos,fams,flags,i,j,methods,flag,flag2,
-      m,nam,val,erg,has,need,isconstructor;
+local oper,opargs,nopargs,verbos,fams,flags,i,j,methods,flag,flag2,
+      m,nam,val,has,need,isconstructor,
+      nummeth,valid,applic,numapplic,nr,first,last;
   if Length(arg)<2 or not IsList(arg[2]) or not IsFunction(arg[1]) then
     Error("usage: ApplicableMethodTypes(<opr>,<arglist>[,<verbosity>[,<nr>]])");
   fi;
+  ## process the arguments 
   oper:=arg[1];
   isconstructor:=IS_CONSTRUCTOR(oper);
-  args:=arg[2];
-  if Length(arg)>2 then
-    verbos:=arg[3];
-  else
-    verbos:=0;
+    opargs:=arg[2];
+  nopargs:=Length(opargs);
+  verbos:=0;
+  if Length(arg)>2 and IsPosInt(arg[3]) then
+    verbos:=Minimum(arg[3],4);
   fi;
-  if Length(arg)>3 then
-    if IsInt( arg[4] ) then
-      skip:=arg[4] - 1;
-    else
-      skip:= -1;
+  nr:=1;
+  if Length(arg)>3 then 
+    if IsPosInt(arg[4]) then
+      nr:=arg[4];
+    elif arg[4]="all" then 
+      nr:=0;
     fi;
-    erg:=[];
-  else
-    skip:=0;
   fi;
-  narg:=Length(args);
-
+  ## accept the name of a function 
+  if not oper in OPERATIONS then 
+    nam:=FilenameFunc(oper); 
```

Comment:
Perhaps also check for the case wher oper is not a function? And add a test for this, too?

Reply:
The check at the start of the ApplicableMethodTypes includes "... or not IsFunction(arg[1]) ..." and an error is issued.  ApplicableMethod contains the lines: 
      elif not IsFunction(arg[1]) then 
           Print("#I  argument 1 must be the name of an operation\n" );